### PR TITLE
adding factorial

### PR DESCRIPTION
**Pull request by jpg-130**
calculating factorial using memoization added

### CODE REVIEW

**Comment by cclauss** [comment]
Could you please add a doctest or two?

**Comment by jpg-130** [comment]
It is showing that the travis-ci build failed, however, it works well on the local machine. What changes should I incorporate?

**Comment by cclauss** [comment]
https://travis-ci.org/TheAlgorithms/Python/builds/558476077#L705
Think like a compiler would.  Is the variable spelled correctly?  Is it in scope?

**Comment by cclauss** [comment]
Need another hint?

**Comment by jpg-130** [comment]
@cclauss I'll try.

File: dynamic_programming/factorial.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+##result[0]=result[1]=1
+##print(factorial(5))
+# print(factorial(3))
+# print(factorial(7))
+
+
+if __name__ == "__main__":
+    import doctest
+    result=[-1]*10
+    result[0]=result[1]=1
```
Comment (cclauss) [inline dynamic_programming/factorial.py]: Hint: Move lines 33 and 34 ___inside___ the __factorial()__ function.

Comment (jpg-130) [reply dynamic_programming/factorial.py]: Ok but if I do that the values are recomputed every time the function is called i.e. I don't want to recompute 3! if 7! is already computed.

Comment (cclauss) [reply dynamic_programming/factorial.py]: Understood.  Then put those two lines at the __top__ of the file at _global scope_ __before the function and the __main()__

Comment (jpg-130) [reply dynamic_programming/factorial.py]: Thank You!

**Pull request merged**